src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -192,7 +192,100 @@ private DenseMatrix initRhs(LfNetwork lfNetwork, EquationSystem equationSystem,
+                monitoredLines.add(branchFactor.getFunction().getBranchId());
+            } else {
+                factors.add(oldFactor);
+            }
+        }
+
+        monitoredLines.forEach(
+            lineId -> {
+                // todo: check that the line is still in the connected component ?
+                BranchFlow branchFlow = new BranchFlow(lineId, lineId, lineId);
+                network.getGenerators().forEach(// todo: not all generators, only those who have an impact in compensation
+                    generator -> {
+                        if (generator.getTerminal().getBusView().getBus() == null) {
+                            return;
+                        }
+                        // check connected component
+                        if (lfNetwork.getBusById(generator.getTerminal().getBusView().getBus().getId()) == null) {
+                            return;
+                        }
+                        String genId = generator.getId();
+                        factors.add(new BranchFlowPerInjectionIncrease(branchFlow,
+                                new InjectionIncrease(genId, genId, genId)));
+                    }
+                );
+            }
+        );
+        return factors;
+    }
+
+    /**
+     * Return a list of id from participating element, and its participation value
+     * @param lfNetwork
+     * @param loadFlowParameters
+     * @param openLoadFlowParameters
+     * @return
+     */
+    private List<Pair<String, Number>> getParticipatingElements(LfNetwork lfNetwork, LoadFlowParameters loadFlowParameters, OpenLoadFlowParameters openLoadFlowParameters) {
+        List<Pair<String, Number>> contributionList = new ArrayList<>();
+
+        if (loadFlowParameters.isDistributedSlack()) {
+            switch (loadFlowParameters.getBalanceType()) {
+                case PROPORTIONAL_TO_GENERATION_P_MAX:
+                    DistributedSlackOnGenerationOuterLoop outerLoop = new DistributedSlackOnGenerationOuterLoop(openLoadFlowParameters.isThrowsExceptionInCaseOfSlackDistributionFailure());
+                    List<AbstractDistributedSlackOuterLoop.ParticipatingElement<LfGenerator>> participatingElements = outerLoop.getParticipatingElements(lfNetwork);
+                    outerLoop.normalizeParticipationFactors(participatingElements, "generator");
+                    participatingElements.forEach(participatingElement -> contributionList.add(Pair.of(participatingElement.element.getId(), participatingElement.factor)));
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Unknown balance type mode: " + loadFlowParameters.getBalanceType());
+            }
+        }
+        return contributionList;
+    }
+
+    /**
+     * Create a map containing sensibility values, indexed first by FunctionId, then by factorId
+     * @param sensitivityValues
+     * @param remainder
+     * @return
+     */
```

Comment:
hardly understandable, need refactoring

Reply:
Indeed, it is the part of the code that is the less understable for me.